### PR TITLE
Adds new functions for Verifier/Signer 

### PR DESCRIPTION
**Pull request by bkchr**
Also introduces new RSA padding and functionality to get the OID of a key.

### CODE REVIEW

File: openssl/src/pkey.rs [hunk 1]
```diff
@@ -70,6 +70,16 @@ pub enum Public {}
 /// A tag type indicating that a key has private components.
 pub enum Private {}
 
+/// The OIDs that identify the type of a key.
+#[derive(PartialEq, Debug)]
+pub enum OID {
```
[the remark this answers is not in the document]
Comment (sfackler) [reply openssl/src/pkey.rs]: Rust naming conventions would call this `Oid`.

**Comment by bkchr** [comment]
@sfackler I integrated your feedback. What do you think of the enum for encoding the special values of the salt length?

File: openssl/src/pkey.rs [hunk 2]
```diff
@@ -155,6 +177,18 @@ impl<T> PKeyRef<T> {
             Ok(EcKey::from_ptr(ec_key))
         }
     }
+
+    /// Returns `Some(Oid)` as the type of this key or `None`, if the Oid is supported by this
```
Comment (sfackler) [inline openssl/src/pkey.rs]: Docs need an update

File: openssl/src/sign.rs [hunk 3]
```diff
@@ -78,6 +79,29 @@ use ffi::{EVP_MD_CTX_free, EVP_MD_CTX_new};
 #[cfg(any(ossl101, ossl102))]
 use ffi::{EVP_MD_CTX_create as EVP_MD_CTX_new, EVP_MD_CTX_destroy as EVP_MD_CTX_free};
 
+/// Salt lengths that must be used with `set_rsa_pss_saltlen`.
+pub enum RsaPssSaltlen {
```
Comment (sfackler) [inline openssl/src/sign.rs]: We'll probably want to keep this extensible - might as well use the same newtype wrapper approach as we did for Oid - cosnts for -1 and -2 and a constructor for custom.

**Comment by sfackler** [comment]
Thanks!

**Pull request merged**